Ui sprint4

Alyssaf21 opened the pull request:
Implemented the backdrop panel and updated the interface for Set Time to match the mockups. Added a new package for the backdrop panel and removed the duration picker package. Moved Save and Settings buttons to the back panel menu. Cleaned up unused code and replaced some hard-coded colors with colors from Theme.dart.

Alyssaf21 committed:
Updated Set Timer screen to reflect mockups

Added space between Ends at: and duration, changed preset button color,
removed Cancel and the top right close button. Changed the am/pm from
Ends at: to be lowercase and reduced the font size.

// import 'package:flutter_duration_picker/flutter_duration_picker.dart';

Alyssaf21 committed:
Removed duration picker package

Alyssaf21 committed:
Changed button colors and modified dark theme

Made Reset button white, changed background color of Presets button,
changed primaryColorLight to be light green on dark mode.

Alyssaf21 committed:
Implemented backdrop panel

Added backdrop package and imported to Layout. Replaced Scaffold with BackdropScaffold, replaced generateAppBar() with getAppbarbuttons() and moved the title, Settings and Save Preset to back panel.

Alyssaf21 committed:
Cleaned up code, applied revisions from merge #15

Deleted unused code and redundant comments from an earlier UI update.
Replaced hard-coded colors with theme colors and added theme.dart as an
import to Pietimer.dart

Alyssaf21 committed:
Merge branch 'master' of https://github.com/elmmo/pie_time into uiSprint4

Alyssaf21 committed:
Fixed file import to match recent merge

elmmo commented on lib/Layout.dart:
```diff
@@ -10,96 +11,112 @@ import 'presets/NewPresetModal.dart';
-                })),
-      ],
-    );
+  List<Widget> getAppbarbuttons(BuildContext context) {
```
Unless you have to override another function with this method signature, use lowerCamelCase

elmmo commented on lib/setup/SetTime.dart:
```diff
@@ -56,16 +56,11 @@ class _SetTimeState extends State<SetTime> {
               buttonPadding: EdgeInsets.fromLTRB(30, 10, 30, 10),
               children: <Widget>[
                 FlatButton(
-                    child: Text("Cancel", style: TextStyle(fontSize: 20)),
```
Mixed feelings on taking out cancel and exit but it doesn't change the functionality of the app, just makes the flow more specific. If this is the way you think we should go, let's do it

elmmo commented on lib/setup/SetupController.dart:
```diff
@@ -44,21 +44,6 @@ class SetupController {
   AppBar getSetupAppBar(String titleText, {Function closeCallback}) {
     return AppBar(
       title: Text(titleText), 
-      actions: <Widget>[
```
Same comment as up above concerning the removal of cancel and exit

Alyssaf21 committed:
Implement suggestions from PR #20 review

Fix case on getAppbarButtons(), put exit button back on the set time/add
task appbar, remove confirmation dialog from exit flow on add task
